### PR TITLE
[JENKINS-55449] Add tool to update "@since TODO" entries

### PR DESCRIPTION
**Pull request by batmat**
[JENKINS-55449](https://issues.jenkins-ci.org/browse/JENKINS-55449)
No JIRA. No changelog because fully internal/dev tool.

The second commit is a run of the tool, the output was:

```
$ ./update-since-todo.sh
Analyzing core/src/main/java/hudson/EnvVars.java:91
        first sha: 45c837daa920e85f84f288a7337016a853054add
        first tag was jenkins-2.144
        Updating file in place
Analyzing core/src/main/java/hudson/EnvVars.java:100
        first sha: 45c837daa920e85f84f288a7337016a853054add
        first tag was jenkins-2.144
        Updating file in place
Analyzing core/src/main/java/hudson/Util.java:967
        first sha: beadc5946f4415b28cb6d7a3a9cfb8a2782ab316
        first tag was jenkins-2.144
        Updating file in place
Analyzing core/src/main/java/hudson/cli/DisablePluginCommand.java:39
        first sha: 1062f9abd86271bfec678d4f1f103639d11873ca
        first tag was jenkins-2.151
        Updating file in place
Analyzing core/src/main/java/jenkins/model/SimplePageDecorator.java:63
        first sha: 94076ad93c13b4098cc6450676baa1bd425c3d32
        first tag was jenkins-2.156
        Updating file in place
Analyzing core/src/main/java/jenkins/util/xstream/SafeURLConverter.java:41
        first sha: 727d58f690abf64f543407e1de3545eca76ad30e
        first tag was jenkins-2.121.3
        Updating file in place
```

### Submitter checklist

- ~JIRA issue is well described~
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- ~Appropriate autotests or explanation to why this change has no tests~
- ~For dependency updates: links to external changelogs and, if possible, full diffs~

### CODE REVIEW

**Review by jvz — APPROVED** [review]
This is pretty cool. It can even be adapted to other repositories to be able to go back and add all the missing `@since` tags.

**Review by daniel-beck — APPROVED** [review]

File: core/src/main/java/jenkins/util/xstream/SafeURLConverter.java [hunk 1]
```diff
@@ -38,7 +38,7 @@
  * for {@link URLStreamHandler#equals(URL, URL)} or {@link URLStreamHandler#hashCode(URL)}. 
  * Required to protect against SECURITY-637
  * 
- * @since TODO
+ * @since 2.121.3
```
Comment (daniel-beck) [inline core/src/main/java/jenkins/util/xstream/SafeURLConverter.java]: This is probably wrong / needs to take into account LTS backports (or as in this case, restricted security fixes).

Comment (batmat) [reply core/src/main/java/jenkins/util/xstream/SafeURLConverter.java]: Not sure what you mean @daniel-beck. I was also originally surprised by it too, but checking manually `2.121.3` seemed indeed like the first version where this got introduced. I.e. was *not* a backport AFAICT.

Or are you saying we should probably not ever put LTS version there, but only the first weekly where a change landed in?

**Comment by batmat** [comment]
Given the three approvals, I plan to merge this before the end of week. Please anyone feel free to weigh in in the meantime. 

Thanks a lot for the reviews @jvz @daniel-beck @kzantow!

**Comment by batmat** [comment]
Merging given positive reviews. Anyway, it's easily addressable if anything surfaces afterwards, and not ever used for user-facing concerns.

File: core/src/main/java/jenkins/util/xstream/SafeURLConverter.java (hunk 1 above)
Comment (daniel-beck) [reply core/src/main/java/jenkins/util/xstream/SafeURLConverter.java]: @batmat I checked the script and you're looking for the oldest tag. If something is introduced in a security update, the LTS is built and tagged an hour or two before the weekly.

Comment (daniel-beck) [reply core/src/main/java/jenkins/util/xstream/SafeURLConverter.java]: @batmat Would be a clear improvement over what we have, so +1 for that suggestion.

Even fancier: Obtain both LTS and non-LTS, compare LTS baseline to non-LTS, include both if baseline older than non-LTS. (Perhaps with added `TODO: Check whether @Restricted in LTS`).

**Pull request merged**